### DISCUSSION TITLE
Full relay with discord and irc

### DISCUSSION THREAD

**Discussion by BCC-BGChatCommunity** [qaisjp/go-discord-irc#119] [Ideas]
My idea is to make it all relay bot with all modes and everything if it is possible. When someone g-line some one of discord puppets discord bot ban user from server . Two side puppets in discord and irc. Every user in irc to be generate in discord server to make possible to message exact user in list. 
If this possible? 
Thanks!

**Comment by qaisjp** [comment]
> g-line

Yes, this is possible. But it is not implemented at all.

> Every user in irc to be generate in discord server to make possible to message exact user in list.

Yes, this is possible. But it is not fully implemented. Right now every user expires after 24 hours (or whatever time is set in the config). You need to add code that will treat a timeout duration of `-1` as "never disconnect", and to also use that same setting to always connect all puppets when the bridge starts.

Right now, when the bridge starts, it only creates puppets if someone is online (or if they come online later).

**Reply by BCC-BGChatCommunity** [reply]
Thanks for the answer. For second part of my idea i mean something else. I know that for only online users is active and this is great but i thing for two side puppets generate  not only in IRC side. Can discord server generate users inside discord and in irc 2 side puppets.First time in irc and second in discord.

**Reply by qaisjp** [reply]
> Can discord server generate users inside discord 

This is not possible:

1. The only way to show additional @-able users is to programmatically create an actual account (against ToS) or a bot
2. It is not possible to automate bot user creations, and if it was, it would not be scalable to go through such a process for every user that joins.
